### PR TITLE
An attempt to generate SQL statements from json-like dto on Colloboque side

### PR DESCRIPTION
**Pull request by edgarzhavoronkov**
Client side is in progress, but a review is highly appreciated

### CODE REVIEW

File: ganttproject/src/main/java/net/sourceforge/ganttproject/storage/Xlog.kt [hunk 1]
```diff
@@ -29,6 +29,40 @@ data class InitRecord(
...
+@Serializable
+sealed class OperationDto() {
+  @Serializable
+  data class InsertOperationDto(
+    val tableName: String,
+    val values: Map<String, String>
+  ): OperationDto()
+
+  @Serializable
+  data class UpdateOperationDto(
```
Comment (dbarashev) [inline ganttproject/src/main/java/net/sourceforge/ganttproject/storage/Xlog.kt:41]: I think we need a condition for update as well. We use task uid for that.

Comment (dbarashev) [reply ganttproject/src/main/java/net/sourceforge/ganttproject/storage/Xlog.kt:41]: Add the processing of these conditions in ColloboqueServer::applyOperation ?

Comment (edgarzhavoronkov) [reply ganttproject/src/main/java/net/sourceforge/ganttproject/storage/Xlog.kt:41]: Yup, forgot to include file in commit

**Review by dbarashev — APPROVED** [review]
LGTM
I think we need some tests to check that json translates to some expected SQL

**Comment by edgarzhavoronkov** [comment]
Ok, i finished generating and sending it on the client side, some debugging and i'll add tests then

**Pull request merged**